### DISCUSSION TITLE
Does this plugin support adding `translate="no"` to HTML elements to avoid browser re-translation?

### DISCUSSION THREAD

**Discussion by blakegearin** [intlify/vue-i18n#1724] [Q&A]
I haven't been able to find an option for this, but searching for it is quite difficult given the keywords associated with it.

Documentation: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/translate

**Comment by JeffreyArts** [comment]
Could you explain me the situation what causes the desire for this? You can just not use {{$t("brandname")}}, and use "brandname" directly

**Reply by blakegearin** [reply]
I'm not sure what you mean.

Situation: I have partial coverage for Spanish translations on a page with gaps in English. If someone uses a browser configured with Spanish as their primary language, the browser will offer to translate the page. When it translates the page, it turns the English to Spanish as expected, but also overwrites the Spanish text. If I have translations, I don't want the browser to interfere with them, which is why I'd want to add `translate="no"` to their HTML elements.

**Reply by JeffreyArts** [reply]
Can you show some code?

If you don't want a piece of text translated, don't wrap it in an i18n wrapper

**Reply by blakegearin** [reply]
Oh, I understand your previous comment now.

The browser doesn't know or care if text went through `v-t` or `$t`. It will translate all text in the HTML that doesn't have `translate="no"` applied.

For example:

```html
<!-- My English translation-->
<span>Application submitted at January 17, 2024 at 12:37 PM</span>

<!-- My Spanish translation -->
<span>Solicitud presentada en 17 de enero de 2024, 12:37</span>

<!-- The browser's override on my existing Spanish translation -->
<span>Solicitud en 17 de enero de 2024, 12:37</span>

<!-- The browser doesn't change or override this -->
<span translate="no">Solicitud presentada en 17 de enero de 2024, 12:37</span>
```

As a point of clarification, each browser has a different translation layer/implementation. [Firefox](https://support.mozilla.org/en-US/kb/firefox-translations-add-on) is local, Chrome is cloud-based, [Edge](https://support.microsoft.com/en-us/topic/use-microsoft-translator-in-microsoft-edge-browser-4ad1c6cb-01a4-4227-be9d-a81e127fcb0b) brands it as Microsoft Translator, etc. 

So the above example was tested in Firefox and may or may not be re-translated by other browsers. But `translate=no` is in the [HTML spec ](https://html.spec.whatwg.org/multipage/dom.html#the-translate-attribute), so it should be adhered to by all.

**Reply by JeffreyArts** [reply]
So the set-up would look something like this:
```
{
 "en": {
     "test": "Application submitted at January 17, 2024 at 12:37 PM"
 },
 "sp": {
     "test": "Solicitud presentada en 17 de enero de 2024, 12:37"
 }
}
```


And this is how the vue template would look like:
```
<span translate="no">{{ $t('test')}}</span>
```

Do I understand your situation correctly that your problem is that the Spanish translation is being translated twice?

**Reply by blakegearin** [reply]
Right, so I don't think `$t` would be able to address the issue. But in theory I think `v-t` would be able to conditionally insert `translate="no"` into the HTML.

The issue with manually adding it in like you showed is that you don't actually want `translate="no"` present when there's gaps in your languages.

So for example:

```json
{
 "en": {
     "test": "Application submitted at January 17, 2024 at 12:37 PM",
     "hello": "Hello"
 },
 "sp": {
     "test": "Solicitud presentada en 17 de enero de 2024, 12:37"
 }
}
```

Then:

```html
<!-- Good because it has full coverage -->
<span translate="no" v-t="'test'" />

<!-- Bad because when displayed in Spanish there's no value -->
<!-- Therefore we do want the browser to be able to translate it from the fallback -->
<span translate="no" v-t="'hello'" />
```

**Reply by JeffreyArts** [reply]
Okay, so the situation is as followed:

You have a page that has some meta data to tell the browser that it is in English. Therefor another feature/plugin of the browser offers to translate it to Spanish. Some of your texts are in Spanish, some are in English. The goal you’d like to achieve here is to add a “translate=no” attribute to some elements to prevent it from translating twice.

The challenge you’re facing here is that you’d like to know if you can programmatically add the “translate=no” attribute to elements that are **not** using the (English) fallback locale. 

Is this a correct depiction of the situation?

**Reply by blakegearin** [reply]
Yes, that seems accurate.

**Reply by JeffreyArts** [reply]
Short answer is no, the <v-t> directive does not automatically add an attribute "translate=no" to the element when it is __not__ using the active locale.

Given that the page is being translated twice, it might be a solution to add "[lang=sp](https://stackoverflow.com/questions/4417923/html-meta-tag-for-content-language)" to your html tag. This would define the page as Spanish, so the browser plugin/feature would understand that the page is in Spanish, and this way it would translate English sentences to Spanish.

**Reply by blakegearin** [reply]
Ok.

I'm already setting `lang="es"` on my html tag. Perhaps some browsers respect it, but Chrome and Firefox don't.
